subsys/bluetooth/host/rfcomm.c

```diff
@@ -1429,13 +1429,13 @@ int bt_rfcomm_dlc_send(struct bt_rfcomm_dlc *dlc, struct net_buf *buf)
 	}
 
 	if (buf->len > BT_RFCOMM_MAX_LEN_8) {
-		uint16_t *len;
```

Comment:
I think it would be better to just use net_buf_push to solve this problem instead of trying to trick coverity into understand variable length field in the struct definition. Which just complicates the code, and makes both the compiler and coverity suspicious of the types involved:

```
	net_buf_push_u8(buf, BT_RFCOMM_SET_ADDR(dlc->dlci, cr));
	net_buf_push_u8(buf, BT_RFCOMM_SET_CTRL(BT_RFCOMM_UIH,
					        BT_RFCOMM_PF_UIH_NO_CREDIT));

	if (buf->len > BT_RFCOMM_MAX_LEN_8) {
		uint16_t len = buf->len - 4;

		net_buf_push_le16(buf, BT_RFCOMM_SET_LEN_16(len));
	} else {
		uint8_t len = buf->len - 3;

		net_buf_push_u8(buf, BT_RFCOMM_SET_LEN_8(len));
	}

```

Reply:
+1 to using this solution instead of the one below.

Reply:
+1

Reply:
@MarkWangChinese can you please update the PR with this suggestion?

Reply:
I am changing the PR, but I find that I need fill the header content with the opposite direction of the rfcomm header fields because the net_buf_push_u8 decreases the buf->data.
The codes is a little strange and is not easy to read, is it good?
	/* length */
	if (buf->len > BT_RFCOMM_MAX_LEN_8) {
		uint16_t len = buf->len - 4u;

		/* Length is 2 byte */
		net_buf_push_le16(buf, BT_RFCOMM_SET_LEN_16(len));
	} else {
		uint8_t len = buf->len - 3u;

		net_buf_push_u8(buf, BT_RFCOMM_SET_LEN_8(len));
	}

	/* control */
	net_buf_push_u8(buf, BT_RFCOMM_SET_CTRL(BT_RFCOMM_UIH,
					BT_RFCOMM_PF_UIH_NO_CREDIT));
	/* address */
	cr = BT_RFCOMM_UIH_CR(dlc->session->role);
	net_buf_push_u8(buf, BT_RFCOMM_SET_ADDR(dlc->dlci, cr));

Reply:
we know that it is not a issue, there is no memory overflow, coverity report it by it's logic. Can we just ignore this coverity and only fix the potential unaligned memory writting issue?

Reply:
The coverity report wasn't for the unaligned access though. It was for overwriting the size of the struct through a cast.
If we keep doing that, but in a different way then likely coverity will generate a new report for the new code.

I think using net buf is better, even if you have to push in reverse order.
The other solution would be to have a different header struct with 16-bit length field.

Reply:
OK, updated the PR using net_buf_push.